File: app/src/main/java/ua/syt0r/furiganit/ui/furigana/FuriganaActivity.java
package ua.syt0r.furiganit.ui.furigana;

import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.webkit.WebView;
import ua.syt0r.furiganit.R;

public class FuriganaActivity extends AppCompatActivity {

    public static final String TEXT_EXTRA_FIELD = "text";
    public static final String FURIGANA_EXTRA_FIELD = "furigana";

    public static final String ERROR_EXTRA_FIELD = "error";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_furigana);
        overridePendingTransition(R.anim.activity_start,R.anim.activity_finish);

        String text = getIntent().getStringExtra(TEXT_EXTRA_FIELD);
        String furigana = getIntent().getStringExtra(FURIGANA_EXTRA_FIELD);

        String error = getIntent().getStringExtra(ERROR_EXTRA_FIELD);

        WebView webView = findViewById(R.id.webview);
        webView.loadData(furigana,"text/html; charset=utf-8", "utf-8");
    }

    public void Click(View v){
        overridePendingTransition(R.anim.activity_start,R.anim.activity_finish);
        finish();
    }

    @Override
    public void onBackPressed() {
        overridePendingTransition(R.anim.activity_start,R.anim.activity_finish);
        super.onBackPressed();
    }
}

File: app/src/main/java/ua/syt0r/furiganit/app/furigana/FuriganaActivity.kt
package ua.syt0r.furiganit.app.furigana

import android.app.PendingIntent
import android.content.Context
import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.Observer
import kotlinx.android.synthetic.main.activity_furigana.*
import org.koin.androidx.viewmodel.ext.android.viewModel
import ua.syt0r.furiganit.R

class FuriganaActivity : AppCompatActivity() {

    companion object {
        private const val MIME_TYPE = "text/html; charset=utf-8"
        private const val ENCODING = "utf-8"

        fun startActivityIntent(context: Context): PendingIntent {
            return PendingIntent.getActivity(
                context,
                0,
                Intent(context, FuriganaActivity::class.java),
                0
            )
        }

    }

    private val viewModel: FuriganaViewModel by viewModel()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_furigana)

        viewModel.subscribeOnState().observe(this, Observer { state ->
            when (state) {
                is FuriganaViewModel.State.Loading -> {

                }
                is FuriganaViewModel.State.Loaded -> {
                    webView.loadData(state.html, MIME_TYPE, ENCODING)
                }
                is FuriganaViewModel.State.Error -> {

                }
            }
        })

        closeButton.setOnClickListener { finish() }
    }

    override fun onWindowFocusChanged(hasFocus: Boolean) {
        super.onWindowFocusChanged(hasFocus)
        if (hasFocus)
            viewModel.loadData(intent)
    }

    override fun onStop() {
        overridePendingTransition(R.anim.activity_start, R.anim.activity_finish)
        super.onStop()
    }

}
